purchase_stock_move_line_control/models/purchase_order.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.multi
+    def _create_picking(self):
+        super(PurchaseOrder, self)._create_picking()
+        for order in self:
+            pickings = order.picking_ids.filtered(
+                lambda x: x.state not in ("done", "cancel")
+            )
+            for picking in pickings:
+                tracked_moves = picking.move_lines.filtered(
+                    lambda m: m.product_id.tracking != "none"
+                )
+                for move in tracked_moves:
+                    move.move_line_ids.unlink()
```

Comment:
Try to avoid for loop nests.
```suggestion
            move_lines = pickings.mapped("move_line_ids").filtered(
                lambda x: x.product_id.tracking != "none"
            )
            move_lines.unlink()
```